File: app/src/main/java/com/tangem/di/AppModule.java
package com.tangem.di;

import android.content.Context;

import org.jetbrains.annotations.NotNull;

import javax.inject.Singleton;

@Singleton
public class AppModule {
    private Context appContext;

    public AppModule(@NotNull Context context) {
        appContext = context;
    }

    Context provideContext() {
        return appContext;
    }

}

File: app/src/main/java/com/tangem/di/AppModule.kt
package com.tangem.di

import android.app.Application
import android.content.Context
import dagger.Module
import javax.inject.Singleton
import dagger.Provides

@Module
class AppModule {

    @Provides
    @Singleton
    fun provideContext(application: Application): Context {
        return application
    }

}